sdk/nodejs/automation/cmd.ts

```diff
@@ -37,26 +46,141 @@ export class CommandResult {
     }
 }
 
-const unknownErrCode = -2;
+export interface PulumiCommandOptions {
+    version?: semver.SemVer;
+    root?: string;
+    /**
+     * Skips the minimum CLI version check, see `PULUMI_AUTOMATION_API_SKIP_VERSION_CHECK`.
+     * @internal
+     */
+    skipVersionCheck?: boolean;
+}
 
-/** @internal */
-export async function runPulumiCmd(
+export class PulumiCommand {
+    private constructor(readonly command: string, readonly version: semver.SemVer | null, readonly root?: string) {}
```

Comment:
Is root really optional?

Reply:
I think so, if we don't set root we run `pulumi` with no path and let the normal $PATH lookup machinery find it.